packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -202,11 +469,43 @@ class GrpcGatewayRouteSpec extends GatewayRouteSpec {
               virtualServiceName: this.routeTarget.virtualServiceName,
             },
           },
+          rewrite: this.defaultHostname
+            ? {
+              hostname: {
+                defaultTargetHostname: this.defaultHostname ? 'ENABLED' : 'DISABLED',
+              },
+            }: undefined,
         },
         match: {
           serviceName: this.match.serviceName,
+          hostname: this.match.hostname?.bind(scope).hostnameMatch,
+          metadata: this.match.metadata?.map(metadata => metadata.bind(scope).headerMatch),
         },
       },
     };
   }
 }
+
+function renderHttpGatewayRouteRewrite(scope: Construct, defaultTargetHostname?: boolean, pathRewrite?: HttpGatewayRouteRewrite)
+  : CfnGatewayRoute.HttpGatewayRouteRewriteProperty | undefined {
+  return defaultTargetHostname || pathRewrite
+    ? {
+      hostname: defaultTargetHostname
+        ? {
+          defaultTargetHostname: defaultTargetHostname ? 'ENABLED' : 'DISABLED',
+        }
+        : undefined,
+      prefix: pathRewrite?.bind(scope).prefix
+        ? {
+          defaultPrefix: pathRewrite.bind(scope).prefix?.defaultPrefix,
+          value: pathRewrite.bind(scope).prefix?.value,
+        }
+        : undefined,
+      path: pathRewrite?.bind(scope).path
+        ? {
+          exact: pathRewrite.bind(scope).path?.exact,
```

Comment:
Uhm, no. Like I said before, you should not be repeating expressions with `bind()` parameters.

`renderHttpGatewayRouteRewrite()` should take a `HttpGatewayRouteRewriteConfig`, not `HttpGatewayRouteRewrite`.

Reply:
Addressed